### PR TITLE
refactor(notifications): remove legacy bare-string payload fallback

### LINKED ISSUE #4329: Remove legacy bare-string notification payload fallback
The `_handleNotificationTapPayload` method in `notification_service.dart` has a catch-all fallback that handles legacy bare-string notification payloads (pre-JSON format). This fallback is transitional compatibility code.

## Removal Plan
1. Verify all notification clients (FCM via `_firebaseMessagingBackgroundHandler`, the push notification service) have been emitting JSON payloads with `referencedEventId` + `notificationType` fields for a sufficient rollout period.
2. Add a temporary log line in the catch block to confirm the legacy path is never hit in production before removing it.
3. Once confirmed dormant across one app store release cycle, delete the `catch (_)` block and the `onNotificationTap?.call(payload, null)` fallback.
4. Update the test in `notification_service_test.dart` that covers the legacy path.

## Files
- `mobile/lib/services/notification_service.dart` — `_handleNotificationTapPayload` catch block
- `mobile/test/services/notification_service_test.dart` — `handles legacy bare event-ID payload` test

### PR DESCRIPTION
**Pull request by untreu2**
## Description
The catch block in _handleNotificationTapPayload that handled pre-JSON bare event-ID payloads has been dormant since all notification clients switched to emitting JSON with referencedEventId + notificationType.

Removes the catch(_) block and its corresponding test.

**Related Issue:** Closes #4329 

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: mobile/lib/services/notification_service.dart [hunk 1]
```diff
@@ -479,29 +479,14 @@ class NotificationService {
...
-      }
-    } catch (_) {
-      // Legacy payloads were bare event IDs (plain strings, not JSON).
-      // Treat the whole payload as a referencedEventId with unknown type.
-      // TODO(#4329): Remove this fallback once all clients have been emitting
-      // JSON payloads (referencedEventId + notificationType) for at least one
-      // full app-store release cycle and telemetry confirms this path is
-      // never reached. See issue for the full removal plan.
+    final data = jsonDecode(payload) as Map<String, dynamic>;
+    final referencedEventId = data['referencedEventId'] as String?;
```
Comment (NotThatKindOfDrLiz) [inline mobile/lib/services/notification_service.dart]: This turns a dormant compatibility cleanup into a new uncaught-error path. `jsonDecode(payload) as Map<String, dynamic>` and the following casts can now throw on malformed JSON, non-object JSON, or wrong-shape data, and taps route straight through this method from the plugin callback. I would still remove the legacy bare-string semantics, but keep defensive parsing here: log a warning and return on malformed payloads instead of letting the tap throw.

File: mobile/test/services/notification_service_test.dart [hunk 2]
```diff
@@ -282,20 +282,6 @@ void main() {
       expect(events.single.notificationType, isNull);
     });
 
-    test('handles legacy bare event-ID payload (non-JSON)', () async {
```
Comment (NotThatKindOfDrLiz) [inline mobile/test/services/notification_service_test.dart:285]: Removing the legacy-path test is fine, but this PR should replace it with coverage for the new intended behavior at this parsing boundary. Please add tests that assert malformed payloads do not emit and do not throw, at minimum for: non-JSON payload, JSON that is not an object, and object payloads whose `referencedEventId` is not a string.

**Review by NotThatKindOfDrLiz — CHANGES_REQUESTED** [review]
Removing the legacy bare-string fallback is fine, but this change also removes the only defensive parsing around the local-notification tap payload. Right now malformed JSON, a non-object JSON value, or a non-string `referencedEventId` will throw out of `_handleNotificationTapPayload` and flow into the app-wide Flutter error path. Please keep the legacy behavior removed, but make malformed payloads fail closed with a warning log + early return, and add replacement tests for the invalid-payload cases so this boundary stays hardened.

**Review by NotThatKindOfDrLiz — APPROVED** [review]
Re-reviewed the current diff and prior review threads. The stale requested changes are addressed: legacy bare-string routing is removed, malformed local-notification tap payloads now fail closed with a warning, and replacement invalid-payload tests cover non-JSON, non-object JSON, and wrong-shape referencedEventId cases. I also verified the targeted test file on the PR head with flutter test test/services/notification_service_test.dart.
